packages/web-frontend/src/selectors/dashboardSelectors.js

```diff
@@ -58,12 +59,43 @@ export const selectCurrentInfoViewKey = createSelector(
 );
 
 export const selectCurrentExpandedViewContent = createSelector(
+  [selectCurrentInfoViewKey, state => state.dashboard.viewResponses],
+  (infoViewKey, viewResponses) => {
+    return viewResponses[infoViewKey];
+  },
+);
+
+export const selectIsEnlargedDialogVisible = createSelector(
   [
-    state => state.enlargedDialog.viewContent,
-    selectCurrentInfoViewKey,
-    state => state.dashboard.viewResponses,
+    selectCurrentDashboardGroupIdForExpandedReport,
+    selectCurrentOrgUnitCode,
+    selectCurrentExpandedViewId,
   ],
-  (viewContent, infoViewKey, viewResponses) => {
-    return viewContent || viewResponses[infoViewKey];
+  (dashboardGroupId, organisationUnitCode, viewId) =>
+    !!(dashboardGroupId && organisationUnitCode && viewId),
```

Comment:
Just a general note that I am not really a big fan of using derived data like this to show and hide a modal. I think it is better to explicitly have a piece of state for opening and closing the modal. It is a better separation of concerns (api data vs. app state) and also gives you more control (adding a loader, un-mounting etc). However I think that refactoring this is out of scope of this PR

Reply:
Interesting, thank you for that. Will definitely consider this in future.